sdk/src/types/block/core.rs

```diff
@@ -21,85 +23,134 @@ use crate::types::block::{
 /// A builder to build a [`Block`].
 #[derive(Clone)]
 #[must_use]
-pub struct BlockBuilder {
+pub struct BlockBuilder<B> {
     protocol_version: Option<u8>,
-    strong_parents: StrongParents,
-    weak_parents: WeakParents,
-    shallow_like_parents: ShallowLikeParents,
-    payload: OptionalPayload,
+    inner: B,
     nonce: Option<u64>,
-    burned_mana: u64,
 }
 
-impl BlockBuilder {
+impl<B> BlockBuilder<B> {
     const DEFAULT_NONCE: u64 = 0;
 
-    /// Creates a new [`BlockBuilder`].
+    /// Adds a protocol version to a [`BlockBuilder`].
+    #[inline(always)]
+    pub fn with_protocol_version(mut self, protocol_version: impl Into<Option<u8>>) -> Self {
+        self.protocol_version = protocol_version.into();
+        self
+    }
+
+    /// Adds a nonce to a [`BlockBuilder`].
+    #[inline(always)]
+    pub fn with_nonce(mut self, nonce: impl Into<Option<u64>>) -> Self {
```

Comment:
@thibault-martinez you had the changes already locally to remove it? Do we merge that before this PR here?

Reply:
Ah yeah, let me finish it quickly